src/libraries/System.Text.Json/src/System/Text/Json/AppContextSwitchHelper.cs

```diff
@@ -5,9 +5,13 @@ namespace System.Text.Json
 {
     internal static class AppContextSwitchHelper
     {
-        public static bool IsSourceGenReflectionFallbackEnabled => s_isSourceGenReflectionFallbackEnabled;
+        public static bool IsDefaultReflectionDisabled { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "System.Text.Json.Serialization.DisableDefaultReflection",
+                isEnabled: out bool value)
+            ? value : false;
 
-        private static readonly bool s_isSourceGenReflectionFallbackEnabled =
+        public static bool IsSourceGenReflectionFallbackEnabled { get; } =
```

Comment:
Is there some design doc describing these two new properties?
I'm struggling to understand why we need two of them?

Reply:
IMO new switch should replace old one. New name is more generic but also applies in the old scenario. Old name only applies to specific behavior for us falling back to DefaultJsonTypeInfoResolver

Reply:
The PR is adding just one feature switch, `System.Text.Json.Serialization.DisableDefaultReflection`. 

The `System.Text.Json.Serialization.EnableSourceGenReflectionFallback` switch was shipped in .NET 7 and is documented here: https://learn.microsoft.com/en-gb/dotnet/core/compatibility/serialization/7.0/reflection-fallback#use-an-appcontext-switch. It is compatibility flag and serves a different purpose than the new one.